### PR TITLE
Remove StfBlueprintTrait

### LINKED ISSUE #1513: Remove StfBlueprintTrait
there is only one implementation let's just get rid of it.

### PR DESCRIPTION
**Pull request by jfldde**
# Description

- Removes `StfBlueprintTrait` and implement methods directly on `StfBlueprint`
- Passes concrete `StfBlueprint` type to `CitreaFullnode` and `CitreaSequencer`.

## Linked Issues
- Fixes #1513

### CODE REVIEW

File: crates/fullnode/src/runner.rs
Comment (eyusufatik) [inline crates/fullnode/src/runner.rs:1]: maybe we should go the batch prover way in full node?

Comment (eyusufatik) [reply crates/fullnode/src/runner.rs:1]: wdyt

Comment (jfldde) [reply crates/fullnode/src/runner.rs:1]: In what sense ?

Comment (jfldde) [reply crates/fullnode/src/runner.rs:1]: Fixed in https://github.com/chainwayxyz/citrea/pull/1587/commits/9e4552a50da1c0a992d8f622502ba02d4bcfb565

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**